libimage/runtime.go

```diff
@@ -574,23 +574,30 @@ func (r *Runtime) ListImages(ctx context.Context, names []string, options *ListI
 		options = &ListImagesOptions{}
 	}
 
-	var images []*Image
-	if len(names) > 0 {
-		for _, name := range names {
-			image, _, err := r.LookupImage(name, nil)
-			if err != nil {
-				return nil, err
-			}
-			images = append(images, image)
-		}
-	} else {
-		storageImages, err := r.store.Images()
+	lookupImagesMap := make(map[string]struct{}, 0)
```

Comment:
(“map” is not that useful in a variable name (especially if when one is functionally a “set”, but even a “set” is not all that useful in a name). `[wanted]imageIDs`, perhaps?)